waves/project.py

```diff
@@ -2559,3 +2559,96 @@ def generate_report(
         results_df.index = pd.Index([simulation_name])
         results_df.index.name = "Project"
         return results_df
+
+
+    def generate_report_lcoe_breakdown(self) -> pd.DataFrame:
+        """Generates a single dataframe of all the desired LCOE metrics from the project."""
+
+        # Static values
+        FCR = self.fixed_charge_rate
+        Net_AEP = self.energy_production(units="mw", per_capacity="mw", aep=True, with_losses=True)
+        Net_AEP = Net_AEP / 1000
+
+        try:
+            capex_data = self.orbit.capex_detailed_soft_capex_breakdown_per_kw
+        except:
+            capex_data = self.orbit.capex_breakdown_per_kw
+
+        data = []
```

Comment:
The CapEx calculation can be simplified via the following. The component ordering can also be dropped because the order of the data isn't changing, though let me know if you notice a difference.

```python
turbine_components = ('Turbine', 'Nacelle', 'Blades', 'Tower', "RNA")
financial_components = ('Construction', 'Decommissioning', 'Financing', 'Contingency', 'Soft')
columns = ['Component', 'Category', 'Value ($/kW)', 'Fixed charge rate (FCR) (real)', 'Value ($/kW-yr)', 'Net AEP (MWh/kW/yr)', 'Value ($/MWh)']

df = pd.DataFrame.from_dict(capex_data, orient="index", columns=["Value ($/kW)"])
df["Category"] = "BOS"
df.loc[df.index.isin(turbine_components), "Category"] = "Turbine"
df.loc[df.index.isin(financial_components), "Category"] = "Financial CapEx"
df.Category = df.Category.str.replace("BOS", "Balance of System CapEx")

df["Fixed charge rate (FCR) (real)"] = FCR
df["Value ($/kW-yr)"] = df["Value ($/kW)"] * FCR
df["Value ($/MWh)"] = df["Value ($/kW-yr)"] / Net_AEP
df["Net AEP (MWh/kW/yr)"] = Net_AEP
df = df.reset_index(drop=False)
```

In case you notice an ordering difference, it can be maintained by the ordering logic shown below

```python
...  # column ordering and category definitions from first block

component_order = [*capex_data]  # NOTE: creates the order
df = pd.DataFrame.from_dict(capex_data, orient="index", columns=["Value ($/kW)"])

...  # rest of the capex calculations

df = df.loc[component_order]  # NOTE: reorders the rows in case they were changed
df = df.reset_index(drop=False
```

Reply:
Had to add the following by the end to make it work since the 'Component' column was named 'index':
df = df.rename(columns={'index': 'Component'}) 
Apart from that, it works well. Thanks for the additions.